### DISCUSSION TITLE
Make `ParamMatcher` return a promise

### DISCUSSION THREAD

**Discussion by pierre-H** [sveltejs/kit#8687] [Ideas]
Hello,

Thank you a lot for SvelteKit ! What a project !

It would be great to allow `ParamMatcher` to return a Promise so we could check in database for slugs for example.

**Comment by rgevrek** [comment]
Absolutely aggree on this idea,

trying for headless WP project to differentiate between usual pages for contact form, data privacy etc. and post pages, so that I need to lockup for slugs against an API.

**Comment by eytanProxi** [comment]
@Rich-Harris : is the blocking because ParamMatcher is runed in the browser?
